### ISSUE TITLE
Which service should Reactiflux move to? 

### ISSUE THREAD

**Issue by benigeri** [reactiflux/volunteers#25]
**Update:** Reactiflux is moving to [Discord](https://discord.gg/0ZcbPKXt5bZiQhB5). ([more info](https://github.com/reactiflux/volunteers/issues/25#issuecomment-148891594))

Wondering why we have to leave Slack? Read [this](https://github.com/reactiflux/volunteers/issues/17#issuecomment-147300333).

I will try to periodically update this list to match what everybody is saying. If you would like to add a contender, please format like all of the others. That way I can easy append the list. 

And thanks to @elwayman02 for a comprehensive [comparison chart](http://jhawk.co/team-chat-comparison-chart)
### Discord

Website: http://discordapp.com/
##### Pros
- Amazing product, has been great for chatting so far. Lots of people like it better than Slack.
- Instant invites make it **so easy** for people to join the conversation. Easier than any other service. 
- _Moderation tools will come in very handy._ We're lucky we've had no spam, but that is eventually going to change. I'd rather have moderators than making our community invite only. 
- Free and no limits. 
- Voice chat is awesome. 
- Easy to go from one server to the other, 
- Team has been very open and supportive. Their app is built in React & Flux & React Native
- Hosted
##### Cons
- _Some people might not be able to access it at work._ (Please let me know if you can't access it so that we can try and assess the severity of this.)
- Search on the roadmap, but not yet built
- It could vanish (but they are backed by some of the best VCs in the world, and seem to have good product market fit)
- Having lots of channels get's noisy, and there isn't a way to shortlist channels. There is a very strong likelihood this well be addressed before the end of 2015.
### Gitter

Website: http://gitter.im/

I'm going to get on the phone with the this week. Hopefully they can alleviate some of our concerns. 
##### Pros
- Proven scalability (Free Code Camp has > 50K users)
- No limits for users or search or archive
- Very OSS friendly company.
- Most of us already have GitHub accounts
- Hosted
##### Cons
- Lots of us have complaints about Gitter's apps being sub-par. Site + apps are not as good as Discord
- **Since Gitter channels/rooms are based around Github organizations/repos, it's unclear whether a Reactiflux community should even exist.** See https://github.com/reactiflux/volunteers/issues/17
- I think more of us have Slack open by default than Gitter open by default
- No voice
- No moderation tools
### RocketChat

RocketChat is an OSS Slack clone. Key differentiator is the fact that it is OSS. 
##### Pros
- Seems like it's pretty full featured
- Has voice and video
- OSS / self hosted. Could be a benefit in the long run
- RocketChat can help us migrate slack account
- Lots of ways to sign in (social + username/password)
##### Cons
- Apps are not as polished as Discord or Gitter
- Self hosted, but the RocketChat guys offered to set it up for us
- Several users have encountered issues signing up and logging in
- Some of the mentioned features are still being developed (multiple orgs, slack migration)
### Facebook @ Work

Website: https://www.facebook.com/help/work

**This would be a very different communication model. Instead of channels (chat rooms), we would have a group for each topic. The conversation would happen around posts.**

You can still chat with small groups or individuals, but we lose the notion of big, open public chat rooms. 
##### Pros
- Facebook is obviously a solid, performant product
- Really good search, no limits
- We can have much better medium/long term discussions, for when chat is not sufficient
##### Cons
- This is an early idea and is not definitely possible. We don't have any confirmation from Facebook. 
- We lose big chat rooms. 
### IRC

There's already a #reactjs on freenode.
##### Pros
- At least some React people already use IRC
- Free and no limits
##### Cons
- IRC clients are not great products 
- If you quit your IRC client you will miss out on any mention and DM. So you need an irc proxy (which only power users really use), or you will need to keep the client connected at all times (impossible). 
- Also there is no sync to mobile. 
### Zulip

Website: https://www.zulip.org/
##### Pros
- No limits
##### Cons
- Product is not as good at others
- We need to host it ourselves
- Not being super actively developed
- High switching cost, everybody needs to create new accounts
### Mattermost

Website: http://www.mattermost.org
##### Pros
- No limits, no lock-in
- Looks well designed, product is promising
##### Cons
- Still not as polished as Gitter or Slack
- We need to host it ourselves
- High switching cost, everybody needs to create new accounts

**Comment by petehunt** [comment]
IRC? There's already a `#reactjs` on freenode

**Comment by fubhy** [comment]
There is also ChatGrape. I know the founders and could talk to them about this. https://chatgrape.com/

**Comment by oren** [comment]
:+1: for IRC on freenode

**Comment by robbestad** [comment]
I think IRC is great option, but I desperately need a better IRC client for the Mac. Even mIRC back in the 90s was better than Colloquy is today.

**Comment by frederickfogerty** [comment]
Someone posted a quick poll in the slack room, maybe this should be moved to the first comment, or a new one made. http://strawpoll.me/5715176 Credit @RWOverdijk

**Comment by benigeri** [comment]
@frederickfogerty thanks for the link. Poll is a great idea, but I think it's important to discuss our options a little more before we ask people to vote.

**Comment by bimawa** [comment]
PLS add all channels on IRC mirror

**Comment by frederickfogerty** [comment]
@benigeri Agreed. I'll leave it there just to get initial opinions

**Comment by Daniel15** [comment]
There's always the possibility of using multiple services (eg. IRC for real-time chat and Facebook@Work or  some forum system for long-form posts). Using software to its strengths would probably end up better than a "one size fits all" solution. Additionally, a very large number of open-source communities already use IRC, so the consistency would be good (rather than having IRC for most other communities and something special just for Reactiflux). 

For real-time chat, my vote would be to use IRC on Freenode, along with a nice web UI such as [KiwiIRC](https://kiwiirc.com/) on reactiflux.com for people that don't want to use IRC through their own client. If KiwiIRC doesn't have all the features we like, we could always fork it. It seems to handle large rooms pretty well though.

I was going to add my previous comments from #reactiflux-dev in this Github issue, but Slack doesn't let me scroll back that far, and unfortunately I don't have the history saved in my IRC client (I've been using Slack via IRC this whole time). Classic Slack.

**Comment by benigeri** [comment]
IMO all of the IRC clients, including KiwiIRC are a huge step back from Slack or even Gitter. Happy to be proven wrong if anybody knows of a really awesome IRC client.

**Comment by bimawa** [comment]
I use WeeChat for terminal its openSource and very high uptime.

**Comment by 0rvar** [comment]
The problem is that if you quit your IRC client you will miss out on any mention and DM. So you need an irc proxy (which only power users really use), or you will need to keep the client connected at all times (impossible). Also there is no sync to mobile. IRC is not really an option in my opinion.

**Comment by oren** [comment]
@awestroke $5 a month on digital ocean.  `apt-get install tmux weechat`

**Comment by benigeri** [comment]
@oren it's never that simple

**Comment by bimawa** [comment]
@oren +mosh

**Comment by bimawa** [comment]
MB Admins can create irc bot for its features?

**Comment by nikcorg** [comment]
+1 for IRC

**Comment by Daniel15** [comment]
> IMO all of the IRC clients, including KiwiIRC are a huge step back from Slack or even Gitter. 

That's subjective. Also, the benefit of using an open protocol rather than a proprietary system is that _anyone_ can build on top of it. There's no lock-in. There's a large number of open source IRC clients, and there's also a large number of bots and libraries to write bots. We could automate things in pretty much any way we like. Logs could be collected by a log bot and stored in a database to give anyone searchable history of everything in the channel. Much much nicer than Slack's "history" (if you can even call it that).

IRC is a proven technology, there's a reason it's still popular today. As I mentioned earlier, lots of open source communities already use it, so using IRC would prevent a lot of people from having to use multiple different apps.

> Also there is no sync to mobile

How many people post to Reactiflux from mobile? Do we know?

> The problem is that if you quit your IRC client you will miss out on any mention and DM. 

I'm using ZNC + znc-push + Pushover for this, I get all mentions instantly to my phone. Took a bit of setup but that's just once-off and now it just works and I don't really think about it.

**Comment by 0rvar** [comment]
@Daniel15 You are a power user. If we require people to get their own software stack just to get basic chat functionality, then we exclude a large chunk of users.
I use slack from mobile all the time.

**Comment by Daniel15** [comment]
> If we require people to get their own software stack just to get basic chat functionality, then we exclude a large chunk of users.

For basic functionality, most web UIs are fine. New users just want to chat, and don't care about all the bells and whistles. Basic chat functionality doesn't require users to get their own software stack, it just requires them to visit a web page.

For power users, what's wrong with allowing them to do more advanced things? If I want to use ZNC, what's wrong with that?

**Comment by 0rvar** [comment]
Offline history is a basic requirement for a modern chat

**Comment by 0rvar** [comment]
If there was a good web UI with persistence that could be embedded, I'd be all for it. But "just let all the noobs find some shitty web UI themselves" is not an option.

**Comment by Kevnz** [comment]
IRC
Is it the prettiest? Nope. Is it the shiniest? Nope. Is it the trendiest? Nope. Would there be a limit to how many people can be invited? Nope. Can people access it from any device no matter the platform? Yep.
Honestly, there are some fairly nice irc web clients, and plenty of apps for people to use. Any none irc platform that you do not control is a risk with the size of this community.

**Comment by 0rvar** [comment]
> Honestly, there are some fairly nice irc web clients

Please provide some links. If we go with IRC, we need to have a rock solid path for clueless (as in, clueless wrt IRC) people to follow, not "use whatever"

**Comment by Daniel15** [comment]
> But "just let all the noobs find some shitty web UI themselves" is not an option.

I was suggesting installing a web UI on reactiflux.com that noobs could use, not that noobs should find their own web UI :)

Any particular things you don't like about KiwiIRC? What if it was extended to provide persistent history?

> > Honestly, there are some fairly nice irc web clients
> 
> Please provide some links.

Textual is good on Mac OS. I used to use HydraIRC on Windows which was nice at the time, maybe there's something nicer now though. 10+ years ago I was using mIRC :)

**Comment by 0rvar** [comment]
I vote for gitter. Then people who like IRC can still use that to connect, while everyone else can use the persistent web interface and/or mobile apps

**Comment by pleunv** [comment]
Discord looks suprisingly good... what's the catch?

**Comment by benigeri** [comment]
@pleunv we're not sure yet.

**Comment by RWOverdijk** [comment]
I also wanted to throw this out there: https://irc.gitter.im/

Seeing how currently IRC and gitter seem to be the most liked platforms.. I think combining them is a realistic solution.

**Comment by taion** [comment]
Discord's markdown doesn't support code, and I'd be willing to bet it's not going to scale well to thousands of users.

**Comment by oren** [comment]
@pleunv it's not open source so you might get into similar situation as slack/gitter

**Comment by sgwilym** [comment]
@pleunv @oren It's also targeted at a completely different kind of customer: gamers. Seeing that this situation has arisen out of Reactiflux’s huge community mismatching Slack’s targeted audience of small teams, it seems particularly foolhardy to move to a service so single-mindedly focused on one demographic.

**Comment by orta** [comment]
Might be worth talking to the IRCCloud team too - https://www.irccloud.com

**Comment by jwheare** [comment]
Hi from IRCCloud, let me know if you have any questions about our service.

We have mobile apps for iOS and Android, keep you connected and synced across platforms and we're specifically setting out to solve IRC's bad usability track record.

Onboarding is simple, join the #reactjs channel here: https://www.irccloud.com/invite?channel=%23reactjs&hostname=irc.freenode.net&port=6697&ssl=1 (more onboarding links and badges available [here](https://www.irccloud.com/badges?hostname=irc.freenode.net&channel=%23reactiflux))

You can use IRCCloud for free forever. We currently have an idle disconnect that kicks in if you don't have the app open for 2 hours, but you can pay $5/month to have this disabled. We don't limit backlog, and you can easily download your logs.

Oh and there's drag and drop file uploads too.

Details on pricing and future planned development is here: https://www.irccloud.com/pricing

I suspect pricing will be an issue, but we might be able to sort out a bulk deal. I've seen your very high user numbers but I suspect most aren't active. What's an estimate for your active community size?

**Comment by arikfr** [comment]
@jwheare if I'm on the free plan, and away from my desktop for 2 hours, will I receive push notification for DMs/mentions on the mobile app? When getting back to the desktop client, will I see all history while I was disconnected?

**Comment by jwheare** [comment]
@arikfr after 2 hours, you get disconnected from IRC completely so history stops being logged until you return and reconnect. You won't get notified because you won't be there.

That's the way the limits work currently, but they're not set in stone. We have been internally discussing the possibility of extending the 2 hour window (to e.g. 30 days) but I can't make any guarantees. We need to strike a balance between scaling and a sustainable business model.

**Comment by jwheare** [comment]
Also, bear in mind the #reactjs channel is publicly logged here https://botbot.me/freenode/reactjs/

**Comment by arikfr** [comment]
@jwheare 30 days will be awesome. But I suspect that once you do that, most people will stop paying :)

**Comment by jquense** [comment]
Uh I hate IRC. all clients are still terrible, none that I ve seen support, what I think are the basic requirements for dev chat,  markdown and code formatting/snippets. 

and to be honest the reason I switched to the reactiflux slack was that other IRC channels were aggressively monopolized by d-bags. the neckbearding tends to be much higher there to the detriment of newbies and folks without a long history of IRC involvement

**Comment by ekmartin** [comment]
Discord looks really promising, should get in touch and ask if they intend it to work with this many users.

As for their payment model:

> Discord is always completely free to use with no gotchas. This means you can make as many servers as you want with no slot limitations.
> 
> Wondering how we’ll make money? In the future there will be optional cosmetics like themes, sticker packs, and sound packs available for purchase. We’ll never charge for Discord’s core functionality.

**Comment by jwheare** [comment]
@jquense fwiw IRCCloud has code formatting snippets (pastebins automatically when you send multiline text). Give it a try before writing it off :)

I can't speak for other IRC channels, and I haven't spent a lot of time in #reactjs, but from what I can tell it's pretty civil in there and friendly to newbies.

**Comment by gryzzly** [comment]
:+1: for IRCCloud, it's great. We used it at SoundCloud as internal communication tool + support channel and it worked excellent.

**Comment by LeoFas** [comment]
Hey there, I thought I'd reach out and invite you on www.chatgrape.com (fyi: I'm one of its founders). 

What sets Grape apart is a deeper integration with 3rd party apps and services, ( see, for instance https://tctechcrunch2011.files.wordpress.com/2015/04/well-done.gif?w=1344&h=794 ( a bit outdated, this gif, but it serves to demonstrate part of the integration UI))  with closed sources like GitHub, BitBucket etc, as well as public sources like Giphy, StackOverflow and others. 

We launched ChatGrape as a public Beta in May this year and have some ten thousand active organisations. This was, however, a kind of test-run, to collect user feedback and the like. In the weeks to come, ChatGrape, as it is online now, will be disregarded and re-launched as Grape.io, a more professional enterprise communication solution aimed and catered to the needs and demands of larger organisations, offering features such as Smart Labelling based on natural language processing technologies. We have a number of large enterprises with us already; medium and bigger organisations  are our core business model - in case you were wondering. 

This said, I know a few of the people that are active in the ReactJS meetup group, including members of our team - speaking of which, our entire accessing method of accessing data and information (internally called smart complete) that is displayed in the GIF is built on ReactJS, as are a number of other parts of Grape) 

..so we'd be happy to offer all of you a free premium plan, with all features unlocked and all limits removed. Both on ChatGrape now and seamlessly transitioned to Grape.io once it goes live. 

Let me know if you find this interesting at l@chatgrape.com and we'll set you up with the fee plan.

**Comment by msikma** [comment]
Gitter is a very nice system. It's probably not a huge deal that it's based around specific Github repositories—it's a nice concept, but I'm sure they're just as happy to see a meta group like this one join them.

The only concern I have is that, with the 150+ Slack channels we have right now, will we be able to move all of those over to Gitter or will we be forced in a single or a few rooms only?

I've always got an instance of IRC open, so I wouldn't mind that either, but it still just isn't something you can expect everyone to understand. It's pretty ancient technology. For coders it shouldn't be too difficult to get started with it even with zero knowledge about IRC, but it's still unnecessarily painful. So if we go with this option I feel we'd be putting new people at a disadvantage. Plus, not having offline functionalities (short of a bouncer) is too ancient for my tastes for a community like this one.

PS: would Hipchat be an option?

**Comment by martinklepsch** [comment]
I'm a member of another Slack community and stumbled upon this as it's relevant for us as well.

I recently discovered [Rocket Chat](https://rocket.chat/) which seems to be a worthy contender. As with Mattermost it would require everyone to create new accounts but that seems to be required with Discord and IRC (to some extent) as well.

Rocket Chat also has some integrations for Github etc I believe.

**Comment by fubhy** [comment]
@LeoFas Can you give us a short PROs & CONs list for ChatGrape? We are moving away from Slack because it does not support unlimited members and has a cap on the archived messages essentially rendering large groups impossible. Would that work better with ChatGrape?

**Comment by sanatgersappa** [comment]
Try Telegram Channels - https://telegram.org/blog/channels. It's fast and has clients on every major platform. It also has a bot api that can be used to automate things.

Website: https://telegram.org/blog/channels

Pros

```
Unlimited members
Fast
Has clients on every major platform
Bot API
Can view posts upto day 1 and has permalinks
OSS friendly
```

Cons

```
Needs a Telegram account (free).
```

**Comment by arikfr** [comment]
@sanatgersappa but, if I'm not mistaken, only admins can write on channels. All others can only receive messages...

**Comment by barisbalic** [comment]
DischordApp looks great, but it's also gaming focused, which might mean they end up prioritising things you don't need or want, instead of the things you do.  I think all in all Gitter is the best option on this list.

**Comment by vslinko** [comment]
:+1: for gitter

**Comment by sanatgersappa** [comment]
@arikfr you're right. My bad. Confused it with groups which has its own limitations.

**Comment by chicoxyzzy** [comment]
gitter ftw

**Comment by vramana** [comment]
Kind of :-1: for Gitter. Because it has no search for rooms. You just have manually go through pages and pages to search for a room. 

Even the search inside the particular room doesn't show the all the rooms under an org correctly.

**Comment by vslinko** [comment]
@vramana rooms index could be published on github. [Example](https://github.com/mr-mig/ru-it-chats/blob/master/README.md).

**Comment by LeoFas** [comment]
@fubhy , Sure, first of all, we'd put you on a free premium plan, which means,

**Pros**
- Unlimited messages
- Unlimited searching of messages (in rooms or pm)
- Unlimited integrations
- Unlimited rooms / channels
- Unlimited members
- Deep Service integrations to a range of tools and services already in place (GDrive, DropBox, GitHub, BitBucket, Trello etc)
- Instant access / integration of public sources (YouTube, Wikipedia, Stackoverflow etc)
- Fast
- Native apps for Android and iOS
- Bot API / support in the making, soon to be publicly released
- Permalinks to all messages
- High emphasis on security, encryption etc (data stored off-site on servers in Germany, - Backups in Switzerland)
- IndexAPI to build your own custom integrations (check out our www.indexapi.com )
- Will be accessible via IRC towards the end of the year
- Oauth supported

**Cons**
- Needs a Grape account (free)
- IRC is accessible via hundreds different IRC clients, Grape is - at the moment - only accessible via one 
- Gitter is highly focused on efficiency surrounding GitHub, Grape can do everything Gitter does with GitHub except project management based on GitHub projects but supports many more integrations with direct object referencing (e.g. GitHub issues)
- Bot API not yet ready for the public 

Happy to accommodate you folks - you know where to reach me.

**Comment by jason-braun-sociomantic-mouse** [comment]
I'm 100% anti facebook.  I dont have facebook.  I am not the only one I imagine.  It seems kind of bad to host an open source community in a closed service

**Comment by mattapperson** [comment]
@jason-braun-sociomantic-mouse You do see the irony in that, given that React and Flux are created by Facebook, right? :)

**Comment by JedWatson** [comment]
@jason-braun-sociomantic-mouse not to distract the conversation but doesn't it seem ironic being anti-facebook in this forum given their incredible investment in and stewardship of React.js to date? 

The issue here is we've built a huge community in a closed service whose interests do not (and will not) align with ours. A closed service with aligned interests and long term sustainability isn't necessarily worse than an open one, IMO.

**Comment by jason-braun-sociomantic-mouse** [comment]
@JedWatson @mattapperson  to clarify, i'm against hosting community in a place where it may not be open to all to participate.  I am not anti-facebook, per se

**Comment by JedWatson** [comment]
Ha, @mattapperson beat me to that line by about 15 seconds.

@jason-braun-sociomantic-mouse thanks for the clarification :)

FB@Work (which I assume you're referring to) actually works with specific accounts, not public Facebook profiles. Which is possibly another barrier to entry. Hopefully, wherever we go, we can come up with an account import strategy to ease the process.

I think the biggest downside to FB@Work would be the change in paradigm - there's a lot of value in chat, if we don't have that as part of Reactiflux, it will happen elsewhere.

**Comment by mattapperson** [comment]
I would like to suggest [lets chat](https://github.com/sdelements/lets-chat)

## Let's Chat

Website: http://sdelements.github.io/lets-chat/

**Pros**
- Self hosted (with a nodes / mongo codebase)
- Hubot Friendly
- Multi-lingual
- MIT License if we want to tweak things
- Private rooms should the need arise
- REST based API

**Cons**
- Designed for smaller teams (though works no worse then Slack)
- No mobile or desktop apps
- YACA (Yet Another Chat App to have open)

**Comment by dallonf** [comment]
My vote: anything but IRC. We're a front end web development community, so UX should be a huge part of this decision, and IRC clients (at least free or open source ones) are notoriously [designed by programmers](http://www.uxdesignedge.com/2010/03/dont-design-like-a-programmer/). _(Note to self: start weekend project to build Slack-like IRC client in React + Electron)_

On an unrelated note, it's probably worth adding [Ryver](http://www.ryver.com/) to the consideration. I have no idea how good it is, but it seems to market itself as being everything Slack isn't, including not limiting users or archives.

**Comment by riyadhalnur** [comment]
I would go for Telegram. I've used Gitter in the past. It still has a long way to go

**Comment by jbhatab** [comment]
@dallonf Ryver looks interesting.

Man, Telegram is interesting too.

Looked through the whole thread and Telegram/Ryver stand out the most IMO.

**Comment by zackkitzmiller** [comment]
IRC

**Comment by jbhatab** [comment]
@riyadhalnur Is telegram only for a person though and not a team? More like gmail in terms of function? Then it may not really fill our needs.

**Comment by elwayman02** [comment]
@benigeri Hipchat should be on this list

**Comment by RWOverdijk** [comment]
No it shouldn't. Hipchat is terrible.

**Update:** I use hipchat daily for work. I really don't like it. But that might be because it's atlassian :p

**Comment by jbhatab** [comment]
I feel like we'll experience the same issue with hipchat at scale as with slack.

**Comment by elwayman02** [comment]
Slack is definitely better than Hipchat, but Hipchat is still a great project, and the team has promised unlimited team sizes, unlike Slack.

**Comment by jbhatab** [comment]
unlimited team sizes is pretty interesting for hip chat.

**Comment by mihalik** [comment]
It should be noted that Slack lists unlimited team size just like everyone else.  But, it comes down to supporting a team with 8,000 signups, 1,000 concurrent users, and 10,000 messages every few days.  That causes some performance issues in Slack (try cmd-t in Reactiflux for an example).  We should make sure that wherever we move can actually handle a large team rather than just claiming supporting unlimited team sizes.

**Comment by riyadhalnur** [comment]
@jbhatab Telegram does seem to support chat room but on further investigation, it seems to be limited to 200 people in a room. Ryver seems like a good alternative.

**Comment by jbhatab** [comment]
@mihalik I completely agree. And my gut says things like chatgrape, hipchat, and others will not be cool with reactiflux. Ryver seems like the only that may be fine with it, but I'm not sure if they are even prepared for the scale. Gitter would most likely be fine, right? Maybe?

**Comment by miztizm** [comment]
Discord is looking cool!

**Comment by jbhatab** [comment]
man that looks super super cool.

https://discordapp.com/

and the voice chat. I love voice chat.

**Comment by 3lvis** [comment]
What about https://github.com/Nordaaker/convos? IRC + super powers.

**Comment by dallonf** [comment]
@jbhatab I get the feeling that Ryver would be forced by their own marketing to accept us as sort of Slack refugees, but the move might kill them :laughing:

**Comment by vjeux** [comment]
I had a chat with the lead dev at Discord and I think that going there is going to be the best outcome for Reactiflux.

I created Reactiflux on Discord, you can join here: https://discord.gg/0ZcbPKXt5bYZVCkR

**But, this is for gamers!?**
This is their primary audience but they have a growing community of game mods/addons developers which have very similar needs than React developers.

**Is it going to die tomorrow?**
Unlikely. They have 14 people full time ( https://discordapp.com/company ), funding and are growing like crazy.

**Why do they care about Reactiflux?**
Their website is built 100% with React, the native iOS app is a mix of React Native and ComponentsKit. They love it so much that they reached out to me 2 months ago out of the blue just to say thanks.

**Is it any good?**
Seriously, try it for yourself. I've never seen such a high quality and fast website in a very long time. I tried many of the other projects mentioned in this thread but they all feel half broken :(

**Will it scale?**
They have similarly sized communities already and it runs fine.

**But, it doesn't support displaying code**
I asked for triple backtick support for codeblocks and better gist rendering. It should be live very soon.

**What else will I miss from Slack?**
- Search is not yet implemented but on the roadmap. Not that our current search on Reactiflux is very useful anyway... Discord doesn't delete anything.
- They have a different API for building bots, we'll need to port the bots we have there. ( Check their community if you are interested https://discord.gg/0SBTUU1wZTXiJKA1 )

**What's better?**
- There's a dark theme!
- Discord is primary marketed as a Voice product. So, people will be able to create voice chatrooms if they want to :)
- You don't need to register to chat, you can only register once you find value.

We should give it a real shot, I think that there's potential for it not only to be a good alternative but a better experience.

**Comment by jquense** [comment]
> But, it doesn't support displaying code
> I asked for triple backtick support for codeblocks and better gist rendering. It should be live very soon.

that and it not being IRC is enough for me :P

**Comment by Daniel15** [comment]
Does Discord have an IRC gateway? It would be good to still be able to connect via IRC (for people that like doing that), and IRC makes it easier to make more portable bots (as opposed to coding something specifically for their API). That's one of the only benefits of Slack over other solutions, the fact they have an IRC gateway.

**Comment by jquense** [comment]
discord is sweet. played with it a little bit.  it's really slick. there isn't an irc bridge but it seems like someone could write one pretty trivially...

**Comment by jbhatab** [comment]
Yup played with discord. Looks awesome. And voice chat?!?!? Woah. Game done changed. 

@vjeux Does it look like they have plans to expand to non gamers in terms of marketing and support? I know you say they will support reactiflux, but I don't like being on platforms that inherently aren't inline with your use.

**Comment by btholt** [comment]
How about [euphoria.io](http://euphoria.io)?

**Comment by jbhatab** [comment]
@btholt Their ui looks so bad and I'd rather just use freenode at that point.

**Comment by ekmartin** [comment]
@jbhatab: They definitely want to expand to other types of communities too, at the moment at least two of their developers are chatting in the #general channel of the [Reactiflux Discord](https://discord.gg/0ZcbPKXt5bYZVCkR).

@jquense and @Daniel15: I ported my Slack/IRC bridge to Discord ([discord-irc](https://github.com/ekmartin/discord-irc)) using their unofficial API (which is apparently okay to use until they release their official one), which bridges channels from Slack to IRC. Not really a personal IRC bridge as Gitter's, but it's setup to connect #general on Discord and #reactiflux on Freenode right now.

**Comment by KyleAMathews** [comment]
And real Markdown support! :100:

**Comment by robbestad** [comment]
Discord looks good :+1:

**Comment by jackcallister** [comment]
Had a really smooth on boarding with discord. Big points for a successful migration.

**Comment by ForbesLindesay** [comment]
Discord looks great.  The two things that I would really like to see fixed before migration are:
1. No support for markdown code blocks (it would be good to get some kind of an ETA on this).
2. No iPad app - gitter has the same issue. Using their iPhone app on an iPad is not a good experience.  I'm most likely to have time to engage in conversation on reactiflux when travelling so I'm more likely to be on an iPad than a laptop.
3. All channels un-muted by default.  This could make it kind of noisy.  Ideally I'd prefer the slack model of joining channels one at a time.

On the other hand, I'm loving:
1. Super fast and easy on-boarding
2. Really easy switching between servers (this was a pain on slack)
3. Looks like a good iPhone app

I do really like the look of rocket.chat but it looks like there currently is only one server, kind of public slack taken to the extreme.

**Comment by luisrudge** [comment]
Is someone in contact with folks at discord? Emojis are ugly and the thin font sucks on chrome@windows.

![image](https://cloud.githubusercontent.com/assets/941075/10456623/ac18efd2-7198-11e5-8741-387d717123dc.png)

**Comment by fubhy** [comment]
I am sure that the looks of the emojis are their primary concern. _cough_ ;)

**Comment by luisrudge** [comment]
it should be :D

**Comment by jorishermans** [comment]
What about https://gitter.im? Can that be a great place ...

**Comment by felixakiragreen** [comment]
Loving Discord so far. The dev community is fantastic there.

**Comment by engelgabriel** [comment]
Guys, I can create a https://reactiflux.rocket.chat for your community if you want to give Rocket.Chat a try?

**Comment by ForbesLindesay** [comment]
@jorishermans Gitter's iOS app is completely broken (notifications and unread counts never clear) and they still lack an iPad app :( other than that they seem ok.

@engelgabriel :+1: for reactiflux.rocket.chat, but how do you then connect to that from the mobile apps?  I didn't see any option in them for entering the server address.

**Comment by wyze** [comment]
A con for Discord that others might run into as I did, was it is blocked by Websense at work. So if others are using it at work, where Slack may have been allowed, Discord may be blocked due to it being "Gaming Related".

I was able to switch to a less restrictive network that didn't have websense/firewall/content blocker and get on Discord fine. It looks very nice.

**Comment by elwayman02** [comment]
@wyze are the desktop/mobile apps also blocked by websense?

**Comment by wyze** [comment]
@elwayman02 Yes. The mobile one goes through if I disable my wifi and just use my data connection.

**Comment by ChrisCinelli** [comment]
I still wonder why somebody haven't built an extension of the IRC protocol (IRCplus?) that is back compatible to IRC to enable more modern features a la Slack. 

Anyway, IRC works. No experience with Gitter yet but it looks good. In the end let's pick one and let's make sure it is clear what the choice is.

**Comment by af** [comment]
Discord looks nice– like @Daniel15 my concern is that there isn't an official IRC bridge (which is how I use Reactiflux slack). It looks like this kind of thing [is on their radar](https://twitter.com/discordapp/status/647994542960979970) though

**Comment by Daniel15** [comment]
@ChrisCinelli 

> I still wonder why somebody haven't built an extension of the IRC protocol (IRCplus?) that is back compatible to IRC to enable more modern features a la Slack.

What features? IRC already supports extension in some ways (eg. DCC and CTCP), see the list of extensions at https://github.com/grawity/irc-docs/blob/master/extensions.md. The main issue is getting client support.

**Comment by engelgabriel** [comment]
@ForbesLindesay we have another project where you can indeed enter multiple server address on the mobile apps, we just didn't have the time to fully test and publish. But we can re-prioritise it for you.

**Comment by Daniel15** [comment]
I'm worried about replacing one proprietary closed system with another. What if the same thing happens with Discord in the future? At least with IRC you can move to another server without users having to learn a completely new app or interface.

@ChrisCinelli 

> I still wonder why somebody haven't built an extension of the IRC protocol (IRCplus?) that is back compatible to IRC to enable more modern features a la Slack.

What features? IRC already supports extension in some ways (eg. DCC and CTCP), see the list of extensions at https://github.com/grawity/irc-docs/blob/master/extensions.md. The main issue is getting client support.

**Comment by ForbesLindesay** [comment]
@engelgabriel Solving that issue seems like the primary problem with getting started using rocket.chat right now.  Ideally I'd like some kind of unified sign on followed by easy switching between servers (kind of like discourse does well and slack does very very badly).

@Daniel15 I share your concerns regarding a closed source system here, it seems like we have to be very sure that whatever solution we go with is:
1. Well funded and sustainable.
2. Actively interested in supporting our use case.

Open source can be a good way of ensuring these requirements are met.

However, as much as I love IRC Cloud (it's one of my favourite web services and incredibly cheap considering) it does not have anywhere near the usability level required when posting snippets of code and editing comments to correct mistakes.  Even with IRC Cloud, it gives you very limited context when you first join the channel, very poor integration with the rest of the eco system of apps, and very limited search functionality.

**Comment by elwayman02** [comment]
I've been working on an equal-weight comparison of some team chat platforms out there. The free features are the only ones considered:

http://jhawk.co/team-chat-comparison-chart

Table Reference:

Y - Yes, it is supported
N - Not supported
P - Partial/Below-Average Implementation
S - Coming Soon
PS - Partial Implementation, with Improved Implementation Coming Soon

The scoring system gives each feature equal weight, which is obviously not how you would actually compare these platforms, just something for me to have an idea as to how well-supported the features are across the board.

Scoring:

Y - 1
P - .5
S - .25
PS - .75

Don't take the scores as a definitive measure of the better platform.

RE: IRC -

I _did_ include it in the comparison, but it was difficult because IRC is simply a server protocol, and there are many different clients that can connect. A single user could customize and configure their IRC experience to better maximize the number of available features to some extent (for example, Fuzzy Name Completion), but I instead tried to consider the default, most common IRC experience. That made the most sense, since it considers the average new user trying to utilize the platform. In the case of a unified community being able to take advantage of the necessary features needed for a team collaboration platform to be useful, the fragmentation of clients is a huge downside to IRC, unfortunately.

**Comment by jwheare** [comment]
@ForbesLindesay a few things: 1) can you expand on the usability issues you have with posting and editing pastebins? I'd love to make improvements 2) what sort of context would you like to see when joining a channel? 3) when considering integrations, it's worth bearing in mind the (arguably disorganised) ecosystem of readily available IRC bots out there, and things like Github's built in IRC integration.

@elwayman02 fwiw IRCCloud would score 14 points (history, code, syntax, full upload support, emoji, tab completion, partial integration support through bots and service hooks, only partial OSS). We're also planning features (search, markdown, better channel listing and roles) that would take it to 17, but that's not a fair comparison.

Listing a lowest common score due to fragmentation is a bit like dismissing the Samsung Galaxy S6 and the entire platform because there are so many sub par Android phones out there.

It's clear some people here have historically had negative experiences with IRC, and it may well not be a match for the reactiflux community in the end (with or without IRCCloud), but my motivation is just to present a fair representation of the _current_ state of IRC.

**Comment by elwayman02** [comment]
@jwheare I wasn't specifically considering IRCCloud in my evaluation of IRC, because IRCCloud is a client implementation, not a platform. That said, I'm happy to include IRCCloud as its own client in the comparison, if the suggestion is to recommend IRCCloud as the default client for users of this community to connect to the IRC channels.

My intent is not to put down IRC; I'm being as objective as possible in evaluating features. In fact, you'll note that my first comment in this thread was to suggest HipChat be considered as a Slack alternative, but my comparison ranks it second to last. I'm not trying to play favorites, because I don't know the answer. This sheet was created to help _me_ figure out what my preferred option would be; I'm just sharing it here in case it helps with this discussion as well.

**Comment by Daniel15** [comment]
@elwayman02 - Features like Markdown can be purely client-side, so I don't think it's fair to say "N" for Markdown in IRC. Also, what do you mean by "partial" support for channel listings in IRC? :)

**Comment by Daniel15** [comment]
@elwayman02 - Features like Markdown can be purely client-side, so I don't think it's fair to say "N" for Markdown in IRC. Also, what do you mean by "partial" support for channel listings in IRC? :) 

Also, I don't know how "integrations" differs from "API/bots", as integrations are just API users. The former implies the latter, but for IRC you listed "N" for integrations and "Y" for API/bots.

Anyways, seems we're gravitating towards Discord, which is fine as long as we have confidence in the product.

**Comment by elwayman02** [comment]
@Daniel15 what I mean is that with IRC it's not simple to have an easily accessible list of channels for an org. You would have to write a bot command or some other sort of custom integration, or perhaps host an IRC server yourself and use the server's channel list. With most of these other platforms, it's easy to discover the list of channels you may want to join.

For Markdown I considered (as mentioned previously) the common use-case of client fragmentation not supporting markdown across the board (or at least enough for it to be a regularly available feature)

Integrations I defined as pre-made integrations provided by the platform. For example, Slack & Hipchat both have probably over 100 integrations that can be set up automatically in just a few clicks. API/Bots means there is an open API for building your own integrations/bots for the platform.

@jwheare IRCCloud seems to have more limitations than you stated above, from my testing. History only seems to be personal history, not org/channel history, and since the Free plan will only stay connected for 2 hours after you go idle, there's a lot of history loss going on. It also only seems to have basic emoji support, rather than the full extended support that a lot of other platforms have. And I can't seem to get any sort of code blocks/markdown to work, either.

![___hellfire-house___freenode](https://cloud.githubusercontent.com/assets/1325482/10470250/6b2662d4-71c0-11e5-8f33-9b227265f151.png)

In either case, though, I'm going to add it to the spreadsheet after I've completed my evaluation.

Edit: I found the pastebin integration...that's pretty annoying from a UX perspective, but I guess it's an ok workaround without backtick support.

**Comment by Daniel15** [comment]
> what I mean is that with IRC it's not simple to have an easily accessible list of channels for an org. [...] With most of these other platforms, it's easy to discover the list of channels you may want to join.

You need to actually join before you can view the list though, right? For example, with Slack, you need a team account in order to view the list of channels. How is that different from IRC? In both cases, as long as you're actually in the system, you can  view the list of channels.

Is there some platform that lets you view the channel list before you register or sign in?

**Comment by jwheare** [comment]
@elwayman02 fair points about the free plan, I did mention this earlier in the thread about being open to a pricing discussion:

> I suspect pricing will be an issue, but we might be able to sort out a bulk deal. I've seen your very high user numbers but I suspect most aren't active. What's an estimate for your active community size?

We could for example provide a team account with a set number of free seats, that you could manage for active members, and then less active or drop in members would be able to get by on limited accounts (given they wouldn't have much need for history anyway)

Also, yes, history is per-user, but as mentioned earlier too, public logging of IRC channels is an established art form: https://botbot.me/freenode/reactjs/

Acknowledged on missing custom emoji support, it's something we've considered but not heard that much demand for, so thanks for the data point.

Formatting options are planned too, is that what you meant by "syntax"? What in particular is annoying about the pastebin UX? Note that it's triggered automatically when you try to send multiline or particularly long text, but you can also trigger it manually with the /paste command or from the menu at the right of the input box.

**Comment by elwayman02** [comment]
@Daniel15 sign-in is very different from what I'm talking about. If I login to any of my Slack accounts, I can easily see every channel in that community with just a single click. That doesn't exist for IRC. I have no good way of knowing which channels on that particular IRC server are part of my community and thus should be important to me.

Additionally, Discord actually lets you joined the community via the invite link without needing to register first; you can see the list of channels and interact indefinitely without having to create an account unless you disconnect from the server. They make it very easy to "claim" your account before then, so you can complete the signup process at your leisure rather than having to register right away. So in fact, there are options that avoid the sign-in to view the channel list, but that's _not_ what I was referring to.

@jwheare public logging of IRC channels is fine, but will IRCCloud be able to scroll through those logs and search their history, whether or not the user was online or in the channel at the time? Because that is what a full team chat platform needs to be able to do.

`Syntax` = code syntax highlighting. I broke it out from `Code` because some platforms had support for one but not the other. The multi-step process to create a pastebin is a barrier to usage; for example, Slack supports syntax highlighting via `snippets`, but unless a very large block of code is being sent, nobody uses that feature. Everyone that I've interacted with simply uses the triple backtick syntax to quickly and efficiently create a code block for their message.

**Comment by jwheare** [comment]
@elwayman02 IRCCloud doesn't publicly log, the service I linked to botbot.me lets you scroll through and search. This could be something we integrate directly into the service in future.

**Comment by elwayman02** [comment]
@jwheare :+1:, for me that's a must-have of a good team chat platform.

FWIW, I think IRCCloud looks like an awesome service, and it definitely makes IRC a much-improved experience (I'm saying this as someone who has used IRC for many years). I just don't think that IRC & IRCCloud as built are really focused on the collaborative team chat experience, and it feels like a lot of patches have to applied to make that platform come anywhere close to what these other products can offer in this specific scenario.

**Comment by elwayman02** [comment]
It's worth noting that Code blocks & Syntax highlighting are being worked on for the next Discord release, and while search isn't implemented yet, it's on the roadmap. Discord's score (whatever that is worth) goes from 15.25 to 18 when that happens.

It seems like Discord a solid option, but I'm also impressed by the depth of features Gitter has. That said, the overall user experience of Gitter is less appealing, and their support for channel discovery and community/org groupings is less than ideal.

Rocket.Chat also looks great, but the hosting issue would need to be resolved. The fact that it's open-source is a huge plus, but it lacks support for easy org switching and markdown in chat. However, it does have the best message editing of any platform I've seen...you can use the up arrow to navigate through several messages that you might want to edit, not just the last one. Pretty sweet!

**Comment by jquense** [comment]
the thing the score doesn't capture is that gitter clients are extremely buggy and just flat out often don't work. I like gitters stated feature set but the follow through is just not there.

**Comment by elwayman02** [comment]
The web and desktop clients have been great for me, but I'll admit to not having tested them extensively. I marked Mobile as partial because I've heard their mobile apps are pretty bad, though apparently they are beta testing an Android rewrite currently.

Like I said, the scores don't tell the whole story...it's just a nice basic comparison to get an idea of what features are even supported in the first place.

**Comment by taion** [comment]
I still hit random bugs on the Gitter web app. It also just doesn't feel smooth. Discord feels a lot better.

**Comment by elwayman02** [comment]
:+1: I can agree with that @talon

**Comment by engelgabriel** [comment]
@elwayman02 Rocket.Chat does support MarkDown and Quote:

![image](https://cloud.githubusercontent.com/assets/1000217/10472669/6e2728f4-71f9-11e5-8d65-7352e4e9b821.png)

And Multi-Org is coming in this month:

https://github.com/RocketChat/Rocket.Chat/issues/658

**Comment by elwayman02** [comment]
I meant quote as in quoting people's messages. This is a thing in some
platforms and it's awesome.

**Comment by elwayman02** [comment]
This is great, I'm really glad I posted that spreadsheet. I'm getting a lot
of good information I wouldn't have otherwise known about these platforms,
and I think some of the product reps in this thread are learning what
features they could add to increase value. Awesome!

**Comment by engelgabriel** [comment]
Ok, we have thins planned https://github.com/RocketChat/Rocket.Chat/issues/164

Can you give me a link to the best implementation you've seen so we can use it as a benchmark?

**Comment by engelgabriel** [comment]
@elwayman02 I'm really glad you posted that spreadsheet too.

**Comment by JedWatson** [comment]
@elwayman02 thanks for your work on that. I suspect "The Great Reactiflux Move of 2015" will become valuable community experience across the board, it's very cool to have so many people engaged and helping out.

Both Discord and Rocket.Chat look good to me; As long as we're confident it can scale, I'd tip the scales slightly in favour of RC because it's open source and skimming the code I feel like I could jump in and hack on it pretty easily if I wanted to. (saying this _despite_ the coffeescript, which has never been my personal preference)

@engelgabriel +1 for https://reactiflux.rocket.chat, and if there are any issues maintaining the hosting I'm sure between all of us we can work it out (whether someone sponsors it or we crowdfund)

Getting the migration to happen smoothly for 7k users is (IMO) one of the big issues; How feasible would it be to import our accounts + chat history from slack? we were discussing transforming that data into a publicly searchable (something) before we got the signal from Slack we actually have to move, so we've got access to the full data export and people who can help make that happen.

We can't retrieve passwords but could recreate user accounts and send invite / activation emails to everyone after the migration, which would make the whole experience smoother.

Also re: getting the mobile apps to work with the separate hostname; you mentioned reprioritising that, how much work's involved?

**Comment by engelgabriel** [comment]
We can do the import. We already have the code developed for most of the import from Slack functionality, it is mainly missing the UI. 

We can do it manually for you already, and fire emails with new temp passwords.

The mobile apps with configurable servers have been created on a separated cordova project, and the fully native apps are being developed as new projects too:

https://github.com/RocketChat/Rocket.Chat.Cordova
https://github.com/RocketChat/Rocket.Chat.iOS

I'll give it a higher priority to push the new apps to the stores.

As you can see, we are a very active project, and if you want to know more about it, please see this video:

https://www.youtube.com/watch?v=yzkId54vng8

**Comment by elwayman02** [comment]
I think my main concern with Rocket.Chat is hosting long-term and at that
scale.

**Comment by graywolf336** [comment]
@JedWatson Slack to Rocket.Chat is my highest priority and is something I am currently working on. I have a 20 user Slack instance and am tired of the 10k limit, so I want to move to Rocket.Chat as soon as possible. And not all of them are very technically inclined, so having a smooth transition over from Slack to Rocket.Chat is critical. Your use case has a lot more users thus a lot more data in the Slack export, so your import will probably test the limits of it so I would be curious to test with your data to verify it works with such large data before saying it is production ready.

I can't give a date as to when this import will be finished, but hopefully it will be in beta by the end of next week..maybe even sooner if I can find more time in a day to work on it. I will be happy to provide more details, if you are interested in them.

@elwayman02 If you join the demo of Rocket.Chat, you'll see over 16k users in the General channel. Not all of them are active, but they still exist in the system and it handles it pretty good.

**Comment by geekgonecrazy** [comment]
@elwayman02 What specifically about long term and scale concerns you? Long-Term hosting at scale should be no issue what so ever.  We also have many tools coming to be able to manage data.  And Rocket.Chat also has the ability to easily scale if the load becomes too great.

**Comment by taion** [comment]
Just to re-iterate, the voice support on Discord is really, really cool. Enough that I'd consider using it over Slack, even disregarding the price thing entirely. It wasn't initially something that seemed like a big deal, but in practice it seems to have the potential to be extremely useful.

**Comment by jquense** [comment]
Ya I'd like to reiterate what @taion is saying. the voice chat is sweet.  it adds a lot to the sense of community and interaction that you can seamlessly jump into a voice convo with folks about whatever.

Rocket chat seems cool also but the hosting issue seems like id be a real non trivial concern. has anyone tried chat Grape, we do have an offer for a free premium plan :P

**Comment by taion** [comment]
Possibly more relevant for projects than for something like Reactiflux though. A voice version of #needhelp sounds like my personal vision of hell.

**Comment by geekgonecrazy** [comment]
On the mention of voice chat support.  I'd like to point out that Rocket.Chat also has voice / video chat support.  :)  While it is young it is maturing very quickly.

**Comment by LorbusChris** [comment]
Thank you for making the list and actually testing it all out @elwayman02 thats useful way beyond!
Would you mind adding [Let's Chat](https://github.com/sdelements/lets-chat) for completeness? It was proposed by @mattapperson somewhere in the tl;dr part above

Rocket.Chat looks amazing! Although I just spent way too much time on their website connecting all the dots ;)

+1 for Rocket.Chat or any viable FOSS solution really, despite possible hosting issues. Loving the crowdfunding idea @JedWatson

**Comment by JedWatson** [comment]
@engelgabriel @graywolf336 awesome. thanks :)

@elwayman02 can you expand a bit on your concerns? If we're talking server cost, that's IMO a solvable problem. If there are fundamental technical concerns about the ability of the technology to scale, let's talk specifics. Being OSS it seems relatively safe to me, in that there's visibility and it can be improved over time.

Sounds like if we move to Rocket.Chat, we need to track:
- Server setup
- Slack import + onboarding / invite process
- Mobile app release for custom URLs

At the moment it gets my vote because
- Polished UX
- Viable migration plan
- Looks like it will scale
- OSS / active project and enthusiasm to adopt us
- No "gaming related content" block for some people

Whether it's Rocket.Chat, Discord, or anything else we need to make a call, and organise a working group to lead the transition. @benigeri would you mind laying down the process you'd like to follow to establish consensus? otherwise we can +1 and bike shed forever :grinning:

**Comment by Daniel15** [comment]
@elwayman02 - One major thing missing from your spreadsheet is data portability. Does the service allow data to be exported? This is pretty important if you ever want to migrate to another service in the future, without losing data.

> +1 for Rocket.Chat or any viable FOSS solution really

+1.  At least with open source, you can easily move to another server, or even fork the code if you need to. I think hosting would be less of an issue with an open-source solution vs something that's remotely hosted that you have no control over.

**Comment by chiedo** [comment]
Why not default to IRC until a decision is made about a long term solution? Temporarily using IRC as the solution has no overhead. Just my opinion though!

**Comment by benigeri** [comment]
Wow. Thank you everyone for the huge amount of discussion and feedback. 

I'm going to parse through it as best I can, and update the main issue with a more descriptive shortlist. But in the meantime, it looks like the new set of top contenders are: 
1. Discord (amazing app + already active)
2. RocketChat (OSS, can be hosted, pretty good)
3. Gitter (Github sign in, also pretty good) 

Unless a large number of us are up in arms by this shortlist, let's keep the conversation focused and stop discussing other alternatives. 

There isn't a clear winner yet, so I'd like wait until Friday or this weekend to make a decision. If it's still not clear by then, we'll find a way to reach consensus. Then we can spend the weekend figuring out the transition, and make it official first thing Monday morning.

**Comment by JedWatson** [comment]
:+1: thanks @benigeri!

**Comment by elwayman02** [comment]
My concern with hosting is thinking beyond just one community, honestly.
This is a problem that the OSS community as a whole faces; many communities
have been or are going to be burned by slack. With Rocket.Chat being
self-hosted, I am worried about the long-term hosting solution for multiple
organizations. Obviously, each community should make the choice that works
best for them, but I'm thinking about the broader impact. One of the
benefits of Slack was using an app you already had open for work. It would
be nice for the new home to be one that many organizations could switch to;
then we could have just one app open for many communities still.
Self-hosting makes this less likely to occur, unless Rocket.Chat is going
to try providing hosting for a bunch of OSS communities.

**Comment by benigeri** [comment]
Need to run off will update the shortlist soon. If anyone wants to jump in and help, please do!

**Comment by Daniel15** [comment]
> Gitter (Github sign in, also pretty good)

Github sign in isn't specific to Gitter; Rocket Chat's demo has Facebook, Github, Gitlab, Google, LinkedIn, and Twitter sign-in, as well as username/password :)

> Why not default to IRC until a decision is made about a long term solution? Temporarily using IRC as the solution has no overhead. Just my opinion though!

#reactjs on FreeNode already exists, you can join it right now if you like :+1: 

>  It would be nice for the new home to be one that many organizations could switch to;
> then we could have just one app open for many communities still.

Honestly I think an open protocol such as IRC or XMPP (or failing that, some other open source platform like Rocket Chat) is the only really useful way to achieve that. A remotely-hosted solution isn't going to suit every single community, and leaves the control of the community in the host's hands which results in a single point of failure. What if they want to discontinue the service in the future, or become another Slack? Some people still like being in full control of their platform and data, and using a platform you can self-host ensures you can do just that in the future if needed.

As I write this, FreeNode has ~86000 users online, and ~52000 channels. I could be wrong, but I'm not aware of any open source community has has more concurrent users. 

The benefit of IRC is you can connect to multiple different networks (hosted by multiple different providers) within the same app; you're not tied to any particular provider. It looks like Rocket Chat has the same advantage though. Rocket Chat is looking pretty good.

**Comment by geekgonecrazy** [comment]
@elwayman02 the wonderful thing about that is regardless of what platform they host Rocket.Chat on whether it be a sub domain of Rocket.Chat or any other our clients are going to work with them.  So one client to access multi communities

**Comment by elwayman02** [comment]
All of which have to be self-hosted, which is a bigger barrier to entry than you give it credit for, I think.

**Comment by Daniel15** [comment]
> All of which have to be self-hosted, which is a bigger barrier to entry than you give it credit for, I think.

Is hosting a site really a barrier to entry for a community full of web developers?

**Comment by elwayman02** [comment]
_shrug_ it's not the end of the world, but I know I wouldn't want to deal with it myself. Somebody's gonna have to pay the hosting costs...crowdfunding is a nice idea, but why expect every community to solve this problem when you can get an equivalent experience with a hosted solution?

Regardless, there are pros and cons to each option, and it may very well be that the pros of Rocket.Chat outweigh the cons, but this particular issue _is_ a con...that's all I'm saying.

**Comment by Daniel15** [comment]
>  Somebody's gonna have to pay the hosting costs

That's not really a disadvantage of self-hosted, it's exactly the same with a hosted solution though. _Someone_ needs to pay for the hosting cost somewhere along the line, whether that be via paid users subsidising free users (ie. a freemium model like what Slack does), the people running the community funding it directly, advertising, or someone donating access to the service because they're awesome (as it seems Rocket Chat would like to do). The company is always going to have expenses they need to cover. 

A hosting company can donate a server much like a SaaS company can donate access to their service.

Unfortunately Rocket Chat's site is very basic right now, it doesn't seem to mention system requirements such as amount of RAM used per connection.

> why expect every community to solve this problem 

For what it's worth, a large number of communities already use self-hosted systems, whether that be forum systems, chat systems, bug trackers, whatever.

**Comment by elwayman02** [comment]
:+1: I'm just playing devil's advocate here; I want to make sure we consider all the ups and downs of each option. :)

**Comment by benigeri** [comment]
Considering how much effort it took us to get a a half-assed landing age which was went down multiple times a week, I struggle to see self-hosted as a benefit. 

Sure, in practice, it's nice, but it adds so much overhead. Who fixes it when it goes down? Who has access to the servers. What happens when the maintainer leaves? 

I'm not against self-hosted solutions, but we can't dismiss the challenges of hosting a chat server that manages 1000+ active connections. And if we continue growing at this rate, we'll probably reach 5K or 10K active connections in 1-3 years. It's just not going to be easy to deal with.

**Comment by Daniel15** [comment]
Yeah, I guess a remotely hosted open source system is best in terms of
maintainence. You don't necessarily _have_ to host it yourself, but being
able to do so is definitely a good feature to have. Open source gives you
the choice - either use the remotely hosted version, or host it yourself.
You get the power to switch between the two, rather than being locked in
to a remotely-hosted service forever.
WordPress does this really well for blogs.

> Considering how much effort it took us to get a a half-assed landing age which was went down multiple times a week, I struggle to see self-hosted as a benefit.

Honestly, there's bigger problems with a server if it can't even host a simple landing page :)

**Comment by benigeri** [comment]
@geekgonecrazy some questions for you: 
1. Are there any full time engineers / product people / designers working on RocketChat? 
2. How do you plan to fund future developments of RocketChat? 
3. Is RocketChat used by any communities that have 1K+ peak daily active users?

**Comment by ForbesLindesay** [comment]
I'm really liking rocket.chat here.  It seems to have an incredibly high velocity of development (considering it looks like they only started in May this year).

If they can nail down support for multi-org in their apps I think it's a winner.

As for funding/hosting of rocket.chat: it's MIT licensed, anyone want to build a startup offering "hosted rocket.chat" :smile:?  We could do free for public organisations and charge 20% less for private organisations, and get away with it funding wise by not having to invest in building all the apps.

**Comment by xzyfer** [comment]
It appears this is not the first community to encounter these scaling issues. This article from another community suggest gitter is a solid option.

http://blog.freecodecamp.com/2015/06/so-yeah-we-tried-slack-and-we-deeply-regretted-it.html

**Comment by jquense** [comment]
I am not quite sure how self hosting could be a viable solution here. The main reason we are even in this situation are scale issues, which are definitely not going to be helped by needing to run volunteer OPS on a chat server hosting for thousands of people. Beyond that the idea of crowdfunding is a nice idea but more than likely will fall into the hands of a few dedicated people to pay.

it all seems like **a lot** of hand waving over big issues for the sole benefit of using OSS software. TBH I am not sure why that is being listed as a benefit other than as a philosophical one, practicaly any OSS solution is going to be probably be _less_ stable, consistent, and supported, than a funded business. I understand that folks feel burnt by Slack , but its not like they pivoted suddenly and kicked us off, Large non company teams has never been something that Slack supported, its just a niche that got filled suddenly because the UX of slack is so much better than IRC.

In terms of my specific weigh in I don't like Rocketchat for the big reasons of self hosting, and that there is little chance I will be using it for any other of the communities I am apart of, whereas there is a good chance my work team and other OSS groups may shift to discord for the voice chat.

Also in terms of sign ups, Discord has the nice benefit of not requiring one to chat, this is an under mentioned feature that seems really great for our type of community where folks want to just drop in easily to ask a few questions with no friction.

**Comment by ForbesLindesay** [comment]
@jquense I completely agree that self-hosting is a problem, I would much prefer we went with something that at least had a hosted version available.

I agree that being able to chat without having to signing up would be a significant benefit, but that functionality could definitely be added to rocket.chat (probably for minimal effort).

If we can find some solution that provides a hosted option for rocket.chat that is free for open source use, I can see it being much easier to move lots of other open source communities (e.g. @jadejs) onto.  I think apart from their current lack of a hosted option or sollution for multiple organisations, they look like a far more complete solution than discord (their iPad and iPhone apps look pretty polished, and their onboarding experience is great).

**Comment by jquense** [comment]
I can definitely admit that this may be a matter of personal opinion but I find Rocket chat a good bit less polished than Discord... the Android/ios/windows discord apps are extremely top notch and the voice chat is really excellent. I am not quite sure what you mean by "onboading", but no sign up seems easier than even really easy sign up?

I appreciate the value of using OSS software but I don't think that we should ignore the similar or worse cons b/c of it and it seems like folks are a lot more willing to :wave: past those issues in that case. I think that it is unreasonable to assume that we can just create a broad sustainable meta OSS project with this community overnight, that's a lot for @benigeri and co to shoulder suddenly :)

**Comment by JedWatson** [comment]
@jquense unless I'm mistaken, Rocket.Chat _is_ offering hosting, management, and assistance with the transition.

If cost becomes an issue down the road, I doubt it would be so great that we couldn't deal with it; if not crowd funding (see Redux) then sponsorship is highly likely. There are a lot of generous companies supporting the React community. Because it's OSS we _could_ host it ourselves, but that isn't actually a requirement. (maybe someone else can step in here and clarify...)

Voice chat is also a feature (although I haven't tried it yet) - and I'm not sure that having to sign in to use it is _even a bad thing_. It's a low barrier when you can sign in with GitHub / Facebook / etc.

I'm not just arguing OSS for philosophical reasons. I actually [proposed a community business model to Slack](https://medium.com/@jedwatson/slack-and-oss-communities-130489aae8) a couple of months back (obviously they're not interested)

As @ForbesLindesay said, Rocket.Chat seems to have an incredibly high velocity of development; because it's OSS we can _see_ that. It's not a black box, and that's the complete opposite of our experience with Slack.

I'm not arguing _against_ Discord, btw, just trying to clear things up. They're clearly dedicated to React and based on what @vjeux said, I'd trust them to look after us.

The biggest issue at the moment with Discord is that a number of corporate firewalls block access to it (something something "Game related content")

**Comment by ForbesLindesay** [comment]
Possibly a minor point, but requiring sign up will make it much easier to kick abusive users if that becomes a problem - the effort required to sign up for a new twitter/github/facebook account each time you get kicked is much higher than just opening a fresh browser window and picking a new temporary username.

**Comment by arikfr** [comment]
One dimension that seems to be ignored, is how many of the current community already use one of the solutions?  I mean, how many of us already have Discord, Gitter or RocketChat client already installed and open?

I'm quite sure that most (all?) of us, would love to avoid having to use another chat client.

**Comment by jquense** [comment]
@JedWatson I get where your coming from, and read your proposal, I have to say tho I am _far_ less convinced as you that folks would step up and pay. I agree that we have seen some positive examples of willingness in this particular community, but in general funding communities is an endemic issue in the OSS and the vast majority of cases, it just doesn't happen, even on mission critical projects *cough* Heartbleed. Admittedly that is a whole different question and convesation :p

In the particular here I want to make sure we are applying the same level of scrutiny to Rocket Chat's offers of support and hole filling that we are the close sourced options. Sure anyone _could_ jump in and help out, but who knows the code base? has bandwidth to join a new project? What happens if folks drift away from RC dev, etc, etc. 

Which is to say I am not necessarily arguing against Rocket chat either :) but lets not pretend that being OSS is some magic curall :P 

Also as far as I can tell Rocket.chat does **not** have a hosted option (O unless the are personally oftering it to us?), perhaps other businesses/services offer that but then we are in the hands of two entities, which seems worse...

**Comment by ForbesLindesay** [comment]
@arikfr I'm pretty sure most of our users won't have rocket.chat or discord (although I could be wrong about discord).  I suspect many have gitter, and they do have very responsive support, but currently their mobile apps are severely lacking.

**Comment by taion** [comment]
Unless I missed something, the Rocket Chat voice options consisted of either recording an audio file as an attachment, or 1:1 video calls. There didn't seem to be anything analogous to the Vent/TS-style voice channels on Discord.

**Comment by arikfr** [comment]
@ForbesLindesay exactly, and I think it's a very important factor that will help smoothen the transition.

Maybe @mydigitalself can comment on their plans in regards to mobile apps.

**Comment by ForbesLindesay** [comment]
My understanding is that gitter are currently focused on android, but should have something to release soon.  iOS improvements will be a little further down the line.

**Comment by mydigitalself** [comment]
@arikfr @ForbesLindesay Absolutely. Just by way of background, until quite recently, our team was just 4 people. We've recently raised some more funding and have doubled the team. We now have a dedicated mobile developer and he's shipping new releases every 2 weeks, continually improving the mobile experience. We're the first to hold up our hands and say our mobile experience has sucked a bit.

Right now the focus is Android and will move on to iOS in November.

**Comment by geekgonecrazy** [comment]
> 1. Are there any full time engineers / product people / designers working on RocketChat?
> 2. How do you plan to fund future developments of RocketChat?
> 3. Is RocketChat used by any communities that have 1K+ peak daily active users?

@benigeri
1. Yes there are, as well as increasing outside contributors. 
2. I'll defer this to @engelgabriel founder of Rocket.Chat
3. Yes it is.  @engelgabriel might be able to provide more specific details.

I believe our specific offer is to provide managed hosting to you. So self hosting wouldn't be a concern. In the talk linked a bit earlier [here](https://www.youtube.com/watch?v=yzkId54vng8) at the end Rocket.Chat is compared to wordpress.

We plan to offer managed hosting.  The details are still being worked out.

**Comment by elwayman02** [comment]
A lot of people seem to be mentioning Rocket.Chat's development velocity as
a differentiator... Just for clarity, I'd like to point out that Discord
also launched earlier this year and is still marketed as an alpha product.
They have similarly built an incredibly polished product in a very short
time, and they've been very responsive as a team. I don't think we need to
worry too much about velocity for any of these three products.

**Comment by elwayman02** [comment]
As a side note, I gave Discord the heads up about the corporate filter
problem. I don't know if there is anything they can do, necessarily, but I
figured they should know that it's a problem stopping some people from
wanting to use their service.

**Comment by jhgg** [comment]
Mattermost just hit v1.0, perhaps it would be nice to re-evaluate it! 

http://www.mattermost.org/open-source-slack-alternative-reaches-1-0/

**Comment by Daniel15** [comment]
> practicaly any OSS solution is going to be probably be less stable, consistent, and supported, than a funded business.

"OSS" and "funded business" aren't opposites. Are you saying a closed-source or source-visible JavaScript library would be better than React, purely based on the fact that React is open source? :) 

In fact I think an OSS solution backed by a company is almost always better than a proprietary solution. Anyone can add new features, rather than having to rely on the company's prioritisation of features.

Rocket Chat is a business but their product is open source. Same with Discourse - It's open-source, but they have a remotely hosted version you can pay for (Discourse is pretty terrible as far as forum systems go, but that's a whole other discussion). Offering code for free and having paid hosting and consulting is a viable business model in a lot of cases.

The benefit of open source is that you can use a remotely hosted version _or_ you can host it yourself.  You get the choice. A proprietary solution can **only** be remotely hosted, which limits flexibility and the ability to scale if you ever outgrow the remotely hosted version.

> the UX of slack is so much better than IRC.

Do you mean a particular IRC client? You can't really compare a particular app to a protocol. That's like saying "The UX of Slack is so much better than SMTP" :stuck_out_tongue:

**Comment by jquense** [comment]
> Do you mean a particular IRC client? You can't really compare a particular app to a protocol. That's like saying "The UX of Slack is so much better than SMTP" :stuck_out_tongue:

I think its clear that I meant the clients,can you point to a cross platform client that doesn't loo like it was made in the 90s by a kernel hacker :P

**Comment by jquense** [comment]
> In fact I think an OSS solution backed by a company is almost always better than a proprietary solution. Anyone can add new features, rather than having to rely on the company's prioritisation of features.

provided that company wants to merge PR's for features, that are outside their priorities, which in my experience isn't often a thing that happens often. You can fork sure, but then its not backed by a company, so you lose some of those guarantees.

I don't think OSS and /funded companies are opposites, My generalization was over stated sure. My point is only that OSS can also suddenly die and go away :)

**Comment by geekgonecrazy** [comment]
@jquense this is true.  But a funded company that isn't OSS can just as easily go belly up and take your data with them.

**Comment by vcarl** [comment]
We're getting off in the weeds here. This isn't the place for an OSS vs companies debate, and yes IRC has many clients, but I think it's inoffensive to say that most of them have poor UX, and there isn't a "best" cross-platform option, which increases friction. Given the number of options available for IRC bridges, I think any option we go with will still be accessible via IRC for the die hards.

Even if Mattermost is 1.0 now, they're still self-hosted. IMO it's a non-starter unless they're willing to make an offer as kind as that from Rocket chat (which seems to be against their value proposition).

**Comment by jquense** [comment]
hehe yeah sorry for the tangents :)

**Comment by benigeri** [comment]
Another issue with Mattermost: there are no apps.

**Comment by taion** [comment]
One more nice thing about Discord for this sort of application - no `@all` or equivalent, as far as I can tell.

**Comment by taion** [comment]
Never mind.

**Comment by barrystaes** [comment]
~~never mind i didnt see Zulip was already mentioned and discarded~~

For me making a new account is a non-issue, as i dont really like the way Slack did that per subsite anyway.

**Comment by vjeux** [comment]
Discord now supports codeblocks per our request :)

<img width="363" alt="screen shot 2015-10-15 at 7 04 05 am" src="https://cloud.githubusercontent.com/assets/197597/10516001/01480b46-730b-11e5-918e-03dbdf2ec59b.png">

**Comment by maggo** [comment]
There's an `@everyone` in Discord but you can remove the right to mention everyone

Did you guys create an open Discord server yet?

**Comment by vjeux** [comment]
@maggo: https://discord.gg/0ZcbPKXt5bYZVCkR (we disabled @everyone in the channel)

**Comment by ForbesLindesay** [comment]
From discord's change log:

> **Code Markdown:** Because sharing code should be pretty. This one's for you reactiflux. `console.log('Discord supports inline blocks too');`

**Comment by ForbesLindesay** [comment]
When I post something with a code block in it, it seems to get duplicated, and all other text before the code block doesn't render, and sometimes there is some arbitrary extra text appended after the code block.

Also, editing code blocks doesn't work at all for me.  The edits just don't get saved.

**Comment by elwayman02** [comment]
Probably better to log the bug with them rather than here, lol

**Comment by geekgonecrazy** [comment]
If you guys have any questions about Rocket.Chat please let us know.  

Also feel free to come check out our demo / community chat  - https://demo.rocket.chat to get a good feel for things your self.

**Comment by benigeri** [comment]
**Please, please, please avoid publicizing this announcement until Monday morning, PST.** 

The 62 of us have spent enough time brainstorming, debating, theoreticizating, and rationalizing over the past week. It’s time to make our decision. 

Regardless of whether we settle for Gitter, Discord, or RocketChat, Reactiflux would survive and continue to thrive. React + friends are taking over the world, and there will always be a need for React-related guidance, banter, and discussion. 

**Reactiflux is moving to [Discord](http://discordapp.com/).**

When @vjeux brought up Discord, I was skeptical. How can a gaming chat service be a good replacement for a community like ours? 

But I still clicked the invite link. I saw Discord’s gorgeous invite page. I answered, “What do you want other people to call you?” and clicked continue. A second later I, was in the Reactiflux Discord server, already talking to @vjeux, @brentvatne, @ide, @ekmartin, and @JedWatson.

That was incredible. Joining Reactiflux on Discord was so easy! 

No permission granting, no password, no email confirmation is needed. Not only is Discord an extremely well polished, high performant, and cross-platform chat app, but it also lets new user join Reactiflux more easily than any other chat app. 

Unlike the other services, Discord is already getting traction among the community. 400 of us have already joined, and most of us really, really like it. 

Some other benefits to look forward to: 
- Very competent team that has been very supportive of us joining
- It’s hosted. Less work for us. 
- Really fast and polished cross platform apps, mobile ones too
- Great cross-server support. (A Discord server is like a team in Slack)
- Voice channels, where we can _literally_ talk to each other
- Moderation tools, so we can fight spammers when they come

Nothing is ever perfect, though. Here’s what I’m mainly worried about: 
- Difficult channel management (improvements are being developed) 
- No search (WIP, it will come!)
- A couple of users have Discord firewall at work. This sucks, but it seems to be a very rare case. Currently affects 0.5% of our community (3/400). I hope that these users can get Discord’s domains whitelisted. 

**Please avoid publicizing this announcement until Monday morning, PST.** We want to update the website, set up a channel directory, write up a blog post, and provide some tips before everybody starts talking about it.  

In the meantime, come check out the new Reactiflux: https://discord.gg/0ZcbPKXt5bZiQhB5

I will write up a more detailed post on Monday morning, but I wanted to let all of you know right away. Thank you all for contributing to this conversation. I didn’t even know the majority of the contender services existed until some of you brought them up, including Discord and Rocket.Chat!

Oh, and let me know if you can with the transition, especially with the website.

**Comment by Daniel15** [comment]
^ Here's a more memorable short URL for that long Discord invite link: http://dl.vc/reactiflux  :+1:

**Comment by Daniel15** [comment]
^ I created a more memorable short URL for that long Discord invite link: http://dl.vc/reactiflux (useful at least until reactiflux.com is updated) :+1:

**Comment by vcarl** [comment]
I can help with the website, I already have Heroku permissions. I'll see if I can get something together Sunday.

**Comment by jbhatab** [comment]
We can still use that mockup that scott made but change out the slack image with discord. It was pretty beautiful. We should just change the link to slackin to the discord app.

**Comment by vcarl** [comment]
Link to that @jbhatab?

**Comment by brentvatne** [comment]
Updated the "Getting Help" section of the React Native docs to link to the new Discord invite link! https://github.com/facebook/react-native#getting-help

**Comment by jbhatab** [comment]
@vcarl I'll get a pic as  soon as I can. Just gotta get it from Scott. It's a solid simple template.

**Comment by benigeri** [comment]
@vcarl @jbhatab any serious discussion re: landing page should happen here: https://github.com/reactiflux/www/issues/1

For now, let's do something simple. Just a static html page with:
1. logo
2. link to discord
3. couple lines of text announcing and justifying the switch
4. Eventually a link to the currently non-existing blog post

**Comment by elwayman02** [comment]
For those who care, I've been working on a weighted feature comparison that assigns an importance (weight) to each feature. I update my spreadsheet with a new tab that shows the weighted scores:

http://jhawk.co/team-chat-comparison-chart

For what it's worth, I came up with the weights based on my personal thoughts as to what's important for an OSS community chat; I did not think about what scores each platform would get when I assigned the weights. I only saw the totals afterward.

Additionally, you might note that `Markdown`, `Code`, and `Syntax` each have a weight of 0.5. This is because I felt that they are really part of the same feature which deserves an overall weight of 1.5 (making it more important than most), but for purposes of comparing platforms they were broken into 3 parts since each app supports it differently.

TLDR Top 5 Scores:

```
Discord 20.875
Slack   20.75
Gitter  20
ChatGrape   19.5
Rocket.Chat 19
```

**Comment by Daniel15** [comment]
@elwayman02 - One major thing missing from your spreadsheet is data portability. Does the service allow data to be exported? This is pretty important if you ever want to migrate to another service in the future, without losing data.

**Comment by elwayman02** [comment]
@Daniel15 that opens a whole new can of worms, though. What about data import? What good is being able to export data from a previous service if you can't import it into the new one? Should there even be an expectation of being able to do that? This is a much bigger conversation than I wanted to get into with my spreadsheet.

**Comment by vcarl** [comment]
We have decided, we're moving to Discord. We're putting up new information on reactiflux.com soon.

**Comment by elwayman02** [comment]
@vcarl I don't think anyone is disputing that (I'm all for Discord, personally)...I was just putting up the additional information as reference to those who might want to see it.

**Comment by Daniel15** [comment]
> What good is being able to export data from a previous service if you can't import it into the new one?

Read-only archives, for one.

> Should there even be an expectation of being able to do that?

Export? Absolutely, you should be in full control of your data, including the ability to download a copy of it. Import? Not quite as much of an expectation, but a mature system should support it.

**Comment by namuol** [comment]
What's the rationale to choose Discord over Mattermost?

Seems like as a community of developers, Mattermost is a good long-term bet.
1. It was specifically created to be an alternative to Slack.
2. We can [import our Slack users and archives](https://github.com/mattermost/platform#archiving-and-search).
3. Do we need a feature? We're all developers. [We can actually contribute](https://github.com/mattermost/platform).

**Comment by benigeri** [comment]
@namuol some points about Mattermost
- it lacks a lot of the polish we found in Discord
- doesn't have mobile or desktop apps
- not as easy for new users to join channels
- we'd have to host it ourselves

And Discord had 400+ members before we even settled on it.

**Comment by MoOx** [comment]
Discord looks really nice. Good choice. Nice find :)

**Comment by fczuardi** [comment]
I've been using this great web-based IRC client: http://shout-irc.com/ as a proxy to keep my nickname always connected, it's free software (MIT License) and easy to setup / hack / customize on any box with nodejs.

**Comment by Daniel15** [comment]
@fczuardi Have you tried KiwiIRC or qwebirc at all? I wonder how Shout compares to those. qwebirc is pretty old now, but FreeNode use it as their official web-based client so it's proven to be stable.

**Comment by fczuardi** [comment]
Shout is probably less stable than qwebirc and kiwiirc. I am using it mostly because I like javascript and the default theme.

**Comment by benigeri** [comment]
It's official, Reactiflux has moved to Discord! Join us: http://join.reactiflux.com

Full blog post: https://facebook.github.io/react/blog/2015/10/19/reactiflux-is-moving-to-discord.html

**Comment by benigeri** [comment]
Thanks again to everyone who spent time researching, googling, discussing, arguing, ranting, complaining, suggesting, and contributing. It took us a while, but I'm confident we picked the right platform.

Now we can focus on other things like our website, a searchable directory, migrating people over...

**Comment by LoganDark** [comment]
> You don't need to register to chat

...
